src/ElboDeriv.jl

```diff
@@ -151,6 +166,52 @@ function ElboIntermediateVariables(
 end
 
 
+"""
+Add all the elbo values for an elbo_vars_array in the first element.
+After a value is added, it is cleared.
+"""
+function reduce_elbo_vars_array!{NumType <: Number}(
+    elbo_vars_array::Array{ElboIntermediateVariables{NumType}};
+    num_threads::Int=nthreads())
+
+  if Threaded
```

Comment:
Gotcha.  Rather than delete the test, could you please just change
```reduce_elbo_vars_array!``` to work if Threaded || num_threads > 1?

FWIW, a cleaner way to do the whole thing would be to make Threaded a
variable in ModelParams rather than a global variable in the ElboDeriv
module.

On Tue, Mar 29, 2016 at 1:15 PM, kpamnany <notifications@github.com> wrote:

> In src/ElboDeriv.jl
> <https://github.com/jeff-regier/Celeste.jl/pull/183#discussion_r57792072>:
>
> > @@ -151,6 +166,52 @@ function ElboIntermediateVariables(
> >  end
> >
> >
> > +"""
> > +Add all the elbo values for an elbo_vars_array in the first element.
> > +After a value is added, it is cleared.
> > +"""
> > +function reduce_elbo_vars_array!{NumType <: Number}(
> > +    elbo_vars_array::Array{ElboIntermediateVariables{NumType}};
> > +    num_threads::Int=nthreads())
> > +
> > +  if Threaded
>
> Actually, I'll fix it in the distributed PR.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/jeff-regier/Celeste.jl/pull/183/files/69680690a7ea6670fd54a36e5cb57132184bd8d6#r57792072>
>